app/forms/flood_risk_engine/steps/correspondence_contact_name_form.rb

```diff
@@ -0,0 +1,78 @@
+# Prototype refers to this as 'main' contact which is not that meaningful
+# This is the contact who will be contacted about the activities, to arrange a visit
+# or receive the registration confirmation, so for clarity we refer to this as the
+# Correspondence Contact
+#
+module FloodRiskEngine
+  module Steps
+    class CorrespondenceContactNameForm < BaseForm
+
+      def params_key
+        :correspondence_contact_name
+      end
+
+      def self.factory(enrollment)
+        contact = enrollment.correspondence_contact
+
+        unless contact
+          contact = enrollment.correspondence_contact = FloodRiskEngine::Contact.new(contact_type: :correspondence)
+        end
+
+        new(contact, enrollment)
+      end
+
+      def self.locale_key
+        "flood_risk_engine.enrollments.steps.correspondence_contact_name"
+      end
+
```

Comment:
I don't think an instance is needed to call these methods

 Things is you need an enrollment to construct this class so if you want to use this information in situations - like interrogating locale information, where you don't have or need an enrollment - you can't.

    let(:locale_errors_key) { "#{CorrespondenceContactNameForm.locale_key}.errors" }

 So actually I've been wishing param key was a class method, as like locale key really its class level info, rather than instance level. 

But I'm guessing theirs issues with inheritance chain if params_key was a class method. AFAIK it's  permissible to have both so maybe we can have  
```
      def params_key
        self.class.params_key
      end

      def self.params_key
        :correspondence_contact_name
      end
```